web_poet/fields.py

```diff
@@ -80,48 +80,56 @@ def __init__(self, method):
             self.original_method = method
             self.unbound_method = None
             self.processors: List[Tuple[Callable, bool]] = []
+            self.name: Optional[str] = None
 
         def __set_name__(self, owner, name):
+            self.name = name
             if not hasattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE):
                 setattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE, {})
 
             field_info = FieldInfo(name=name, meta=meta, out=out)
             getattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE)[name] = field_info
 
-        def __get__(self, instance, owner=None):
+        def __get__(self, page, owner=None):
             if self.unbound_method is None:
-                for processor in out or []:
+                if out:
+                    processors = out
+                elif hasattr(page, "Processors"):
+                    processors = getattr(page.Processors, self.name, [])
```

Comment:
Should we look in the parent classes if there is no processor for the field in this one?

Reply:
That’s an option.

Another is to encourage subclassing of processor classes:

```python
    class Page(BasePage):
        class Processors(BasePage.Processors):
            name = [proc1]
```

Reply:
I think we should encourage subclassing, it's more explicit.

Reply:
A couple more examples:

```py
    class Page(BasePage):
        class Processors(BasePage.Processors):
            name = BasePage.Processors.name + [proc1]

        @field(out=BasePage.Processors.description + [str.strip])
        def description(self): ...
```

Reply:
If we encourage this pattern, it could make sense to add a test for it, to ensure it won't break.

Reply:
I've found an interesting case when implementing this: 

```python
@attrs.define
class BasePage(ItemPage):
    class Processors:
        f = [proc1]

    @field
    def f(self):
        ...

@attrs.define
class Page(BasePage):
    class Processors(BasePage.Processors):
        f = [proc2]

p1 = BasePage()
#p1.f
p2 = Page()
#p2.f
```

There is only one `_field` instance and it gets the processor list only once, on the first call of `__get__()`, so `p2.f`  will use the `proc1` processor if `p1.f` was accessed first.

Looks like we either need to compute the processor list on each call or store it in `owner` instead of `self`? In the second option we could use the `FieldInfo` dict in `owner`?

Reply:
The owner approach sounds good to me.